### PR TITLE
Reorganize Test Visibility Java documentation

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### What does this PR do?

Improves Test Visibility documentation for Java:
* documentation sections are reorganised so that concepts that are related (configuration, programmatic extensions, etc.) are grouped together in the same section
* some minor details are added: the list of supported test framework is expanded, a summary of tracer set up steps is provided, etc.

### Motivation

* there was feedback from technical sales people mentioning that documentation lacks clear structure
* recent DD Java Tracer releases added support for more Spock and Cucumber test frameworks

### Additional Notes

There is a `ci-git-metadata` shortcode that is included in Test Visibility documentation for different languages. It outlines Git metadata collection settings and should be the same for every language.
In this PR I have removed the header from the fragment, instead including it in every page that uses the fragment.
I have done this because I felt like in Java docs - due to the way they're structured now - it'd be more appropriate to have the 3rd level header rather than the 2nd level header that is used elsewhere.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/tests/java.md [hunk 1]
```diff
@@ -23,16 +23,29 @@ further_reading:
...
-## Configuring reporting method
+If your test framework is not supported, you can try instrumenting your tests using [Manual Testing API][1].
 
-To report test results to Datadog, you need to configure the Datadog Java library:
+## Setup
+
+To set up Test Visibility for Java, you need to perform the following steps:
+1. Configure tracer reporting method
+2. Download tracer library to the hosts where your tests are executed
+3. Run your tests with the tracer attached
```
Comment (brett0000FF) [inline content/en/continuous_integration/tests/java.md:41]:
```suggestion
1. Configure tracer reporting method.
2. Download tracer library to the hosts where your tests are executed.
3. Run your tests with the tracer attached.
```

**Review by brett0000FF — APPROVED** [review]
Just a heads up for next time: There's no need to edit the alternate language files. Those changes will actually get overwritten when the translation team works through these updates.

Thanks!

**Review by brett0000FF — APPROVED** [review]

**Comment by nikita-tkachenko-datadog** [comment]
@brett0000FF could you please merge this? I do not have the permissions.

**Comment by brett0000FF** [comment]
@nikita-tkachenko-datadog -- You bet! Done 👍

**Pull request merged**